linode/ansible_linode
# new: Support replication in `image` module

## 📝 Description

Allow replicate image by specifying a new field `regions_to_replicate`. User can wait for all replications to finish by specifying `wait_for_replications` and customize `wait_timeout`. 

## ✔️ How to Test

```
make TEST_ARGS="-v image_basic" test
```

Manual Test:
1. In a sandbox environment, i.e. dx-devenv, run the following playbook to create an image and replicate it.
```yaml
- name: test
  hosts: localhost
  tasks:
    - name: Create an instance to image
      linode.cloud.instance:
        label: my-test-ansible-instance
        region: us-east
        type: g6-standard-1
        image: linode/alpine3.19
        state: present
      register: instance_create

    - name: Create an image from the instance
      linode.cloud.image:
        label: my-gen2-image
        disk_id: '{{ instance_create.disks.0.id }}'
        description: 'cool'
        cloud_init: true
        tags:
          - 'image-gen2'
        state: present
      register: image_create

    - name: Replicate the image
      linode.cloud.image:
        label: my-gen2-image
        disk_id: '{{ instance_create.disks.0.id }}'
        description: 'cool2'
        tags:
          - 'test'
        replica_regions:
          - 'us-east'
          - 'eu-west'
        wait_for_replications: yes
        wait_timeout: 1200
        state: present
```
2. Observe that this image is replicate successfully.
3. Try to remove the available regions and put a new region
```yaml
    - name: Replicate the image
      linode.cloud.image:
        label: my-gen2-image
        disk_id: '{{ instance_create.disks.0.id }}'
        description: 'cool2'
        tags:
          - 'test'
        replica_regions:
          - 'us-central'
        wait_for_replications: yes
        wait_timeout: 1200
        state: present
```
4. Expect to see an error.
5. Try to remove one of the available region
```yaml
    - name: Replicate the image
      linode.cloud.image:
        label: my-gen2-image
        disk_id: '{{ instance_create.disks.0.id }}'
        description: 'cool2'
        tags:
          - 'test'
        replica_regions:
          - 'us-east'
        wait_for_replications: yes
        wait_timeout: 1200
        state: present
```
6. Observe that only one available region is returned.
7. Change around the script to discover any edge case.
